tests/loadbalancersim/lb_test.go

```diff
@@ -0,0 +1,378 @@
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package loadbalancersim
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/test/loadbalancersim/histogram"
+	"istio.io/istio/pkg/test/loadbalancersim/loadbalancer"
+	"istio.io/istio/pkg/test/loadbalancersim/locality"
+	"istio.io/istio/pkg/test/loadbalancersim/mesh"
+	"istio.io/istio/pkg/test/loadbalancersim/network"
+)
+
+func TestLoadBalancing(t *testing.T) {
```

Comment:
can we add an env var to enable this (off by default)? Or a build tag - env var is probably easier

Reply:
I've added the build tag and removed the hold